src/xamarin/SkManaged_Png_Chunk_Reader.cpp

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2022 Microsoft Corporation. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+
+#include "include/xamarin/SkManaged_Png_Chunk_Reader.h"
+
+SkManaged_Png_Chunk_Reader::Procs SkManaged_Png_Chunk_Reader::fProcs;
+
+void SkManaged_Png_Chunk_Reader::setProcs(SkManaged_Png_Chunk_Reader::Procs procs) {
+    fProcs = procs;
+}
+
+SkManaged_Png_Chunk_Reader::SkManaged_Png_Chunk_Reader(void* context) {
+    fContext = context;
+}
+
+SkManaged_Png_Chunk_Reader::~SkManaged_Png_Chunk_Reader() {
+    if (fProcs.fDestroy) {
+        fProcs.fDestroy(this, fContext);
+    }
```

Comment:
Just to have 1 line less 😄 

```suggestion
    if (!fProcs.fDestroy) return;
    fProcs.fDestroy(this, fContext);
```